### PR TITLE
Entropy estimators

### PR DESCRIPTION
**Pull request by kzahedi**
Please take a look.

### CODE REVIEW

File: stat/counts.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	for i, v := range c {
+		r[i] = float64(v) / n
+	}
+
+	p := mat.NewVecDense(len(r), r)
+
+	return p
+}
+
+// Counts returns an integer vector with number of occurrences of each number in
```
[the remark this answers is not in the document]
Comment (btracey) [reply stat/counts.go]: Hold off though until I can look at the entropy estimators more closely.

File: stat/it/probabilityestimators.go [hunk 2]
```diff
@@ -0,0 +1,156 @@
+package it
```
Comment (btracey) [inline stat/it/probabilityestimators.go]: We should leave these off this PR since they aren't used here.

File: stat/it/mutualinformation.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
+package it
+
+import (
+	"math"
+
+	"gonum.org/v1/gonum/mat"
+)
+
+// MutualInformation calculates the mutual information with the given lnFunc function
```
Comment (btracey) [inline stat/it/mutualinformation.go]: I think we can just move this into `gonum/stat`. It's cleanly defined, and simple.

**Comment by kortschak** [comment]
Where are we up to with this?

Note that the failure here is due to missing copyright headers.

**Comment by kzahedi** [comment]
The main problem is that I don't understand how to have my own fork of gonum and don't mess up the import paths.

**Comment by kortschak** [comment]
When you locally clone out your repo from your fork don't use `go get` from your fork, use git and put it in the correct path for the gonum project,
```
mkdir -p $GOPATH/src/gonum.org/v1
$GOPATH/src/gonum.org/v1
git clone git@github.com:kzahedi/gonum.git
```

Alternatively, if you already have gonum installed in the correct location, add your fork as a tracked remote to that and work that way (this is how I usually work on third-party projects).

**Comment by vladimir-ch** [comment]
I'm closing this PR because it has gone stale. We're still interested in contributions, so please create a new PR if you still intend to submit this change.

**Pull request closed without merging**